stdlib/public/TensorFlow/Ops.swift

```diff
@@ -1597,52 +1597,315 @@ public extension Tensor where Scalar : Numeric {
+//   public var tensorRange: TensorRange { return ._ellipsis }
+// }
+
+extension Int32 : TensorRangeExpression {
+  public var tensorRange: TensorRange { return ._index(self) }
+}
+
+extension Int : TensorRangeExpression {
+  public var tensorRange: TensorRange { return ._index(Int32(self)) }
+}
+
+extension Range : TensorRangeExpression where Bound == Int {
+  public var tensorRange: TensorRange {
+    return ._range(Int32(self.lowerBound)..<Int32(self.upperBound), stride: 1)
+  }
+}
+
+extension ClosedRange : TensorRangeExpression where Bound == Int {
+  public var tensorRange: TensorRange {
+    return ._closedRange(
+      Int32(self.lowerBound)...Int32(self.upperBound), stride: 1)
+  }
+}
+
+extension PartialRangeFrom : TensorRangeExpression where Bound == Int {
+  public var tensorRange: TensorRange {
+    return ._partialRangeFrom(Int32(self.lowerBound)..., stride: 1)
+  }
+}
+
+extension PartialRangeUpTo : TensorRangeExpression where Bound == Int {
+  public var tensorRange: TensorRange {
+    return ._partialRangeUpTo(..<Int32(self.upperBound), stride: 1)
+  }
+}
+
+extension PartialRangeThrough : TensorRangeExpression where Bound == Int {
+  public var tensorRange: TensorRange {
+    return ._partialRangeThrough(...Int32(self.upperBound), stride: 1)
+  }
+}
+
+public extension Tensor {
+  @_fixed_layout @usableFromInline
+  internal struct IndexPath {
+    @usableFromInline
+    let begin, end, strides: Tensor<Int32>
+
+    @usableFromInline
+    let beginMask, endMask, ellipsisMask, newAxisMask, squeezeAxisMask: Int64
+
+    @inlinable @inline(__always)
+    public init(
+      begin: Tensor<Int32>, end: Tensor<Int32>, strides: Tensor<Int32>,
+      beginMask: Int64, endMask: Int64, ellipsisMask: Int64, newAxisMask: Int64,
+      squeezeAxisMask: Int64
+    ) {
+      self.begin = begin
+      self.end = end
+      self.strides = strides
+      self.beginMask = beginMask
+      self.endMask = endMask
+      self.ellipsisMask = ellipsisMask
+      self.newAxisMask = newAxisMask
+      self.squeezeAxisMask = squeezeAxisMask
+    }
+  }
+
+  @usableFromInline
+  @differentiable(wrt: self, vjp: _vjpSubscript)
+  internal subscript(_ indexPath: IndexPath) -> Tensor {
+    get {
+      return Raw.stridedSlice(
+        self, begin: indexPath.begin, end: indexPath.end,
+        strides: indexPath.strides, beginMask: indexPath.beginMask,
+        endMask: indexPath.endMask, ellipsisMask: indexPath.ellipsisMask, 
+        newAxisMask: indexPath.newAxisMask,
+        shrinkAxisMask: indexPath.squeezeAxisMask)
+    }
+    set {
+      self = Raw.tensorStridedSliceUpdate(
+        self, begin: indexPath.begin, end: indexPath.end,
+        strides: indexPath.strides, value: newValue,
+        beginMask: indexPath.beginMask, endMask: indexPath.endMask,
+        ellipsisMask: indexPath.ellipsisMask,
+        newAxisMask: indexPath.newAxisMask,
+        shrinkAxisMask: indexPath.squeezeAxisMask)
+    }
+  }
+
+  @inlinable @inline(__always)
```

Comment:
```suggestion
  @inlinable
```

We should remove all `@inline(__always)` from this file. `@inline(__always)` is very much an early-phase cruft that was required by GPE. If we keep them here it'll blow up code size during optimizations.